### PR TITLE
[nrf fromlist] wifi: shell: Support configuration of timeout in connect command

### PR DESCRIPTION
**Pull request by rado17**
Add support for configuring timeout in `wifi connect` command.

### CODE REVIEW

**Review by ajayparida — APPROVED** [review]

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -563,6 +565,15 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
...
 				&params->bssid[4], &params->bssid[5]);
 			break;
+		case 't':
+			if (iface_mode == WIFI_MODE_INFRA) {
+				params->timeout = strtol(optarg, &endptr, 10);
+				if (*endptr != '\0') {
+					PR_ERROR("Invalid timeout: %s\n", optarg);
+					return -EINVAL;
+				}
+			}
```
Comment (sachinthegreen) [inline subsys/net/l2/wifi/wifi_shell.c:575]: Should there be an error message if mode is not `INFRA` ?

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (rado17) [reply subsys/net/l2/wifi/wifi_shell.c:575]: We can add a warning log. Since the upstream PR has already been merged, will take this up in a separate PR.

**Review by kapbh — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (sachinthegreen) [reply subsys/net/l2/wifi/wifi_shell.c:575]: Please file a Jira for the same.

**Review by sachinthegreen — APPROVED** [review]

**Comment by rado17** [comment]
Included in Upmerge.

**Pull request closed without merging**